### DISCUSSION TITLE
how to fix  Wsl/WSL_E_WSL_OPTIONAL_COMPONENT_REQUIRED

### DISCUSSION THREAD

**Discussion by CasonPlayzYT** [microsoft/WSL#10074] [Q&A]
it says This application requires the Windows Subsystem for Linux Optional Component.
The system may need to be restarted so the changes can take effect.
Error code: Wsl/WSL_E_WSL_OPTIONAL_COMPONENT_REQUIRED
Press any key to continue... how do i fix that?

**Comment by RoachLin** [comment]
Click the Start button, Control Panel, Programs, Turn Windows features on or off, Hyper-V

**Reply by thymaro** [reply]
except it's not available for Windows 10 Home owners. Will look for another solution.

**Reply by JonasDeSchouwer** [reply]
@thymaro did you find one?

**Reply by thymaro** [reply]
I don't remember. What is the path to get to this error, again? Funnily enough, after this comment, I noticed I had Win10pro and I *could* activate Hyper-V. I didn't, though, but I don't remember whether I found another solution to "the problem", or whether I circumvented it altogether.

**Reply by thymaro** [reply]
@JonasDeSchouwer in [this new comment](https://github.com/microsoft/WSL/discussions/10074#discussioncomment-9345620), @VivekGupta137 might have found out, what my workaround was at the time. The approach feels very familiar.

**Comment by itsarunstark** [comment]
For those who can't find here it is Control Panel -> Programs & Features -> Turn WIndows Feature On or Off -> Windows subsystem for linux and then OK

**Reply by prasaanth2k** [reply]
That also Aready Enabled it not working

**Reply by RogueD13** [reply]
same here

**Reply by ZeddZoul** [reply]
It helped me. I remember turning this off before. thanks

**Comment by parekhsandip** [comment]
This error generally comes immediate after command `wsl -- install` is finished. However, it requires system restarts and then actual ubuntu installation starts.

**Comment by VivekGupta137** [comment]
for me Control Panel -> Programs & Features -> Turn WIndows Feature On or Off -> Windows subsystem for linux was already checked. but wsl still showed same thing after multiple restarts.

what worked for is to first uncheck the "Windows subsystem for linux" then restart, then again check "Windows subsystem for linux" and restart.
retry the wsl.exe --install command and it worked.

mine is windows 10 home

**Reply by thymaro** [reply]
now, that you mention it, I think I did something similar. All in the spirit of "please turn it off, and on, again". It seems dumb, in a case like that, but it seems to be resetting some things with this method.

**Reply by sidnas** [reply]
same here.. after unchecking, restarting and checking back everithing worked

**Reply by Kaiaaaaaaaa** [reply]
This procedure fixed it for me, thank you!

**Reply by edwin** [reply]
this is work for me

**Reply by liamhavoc** [reply]
This worked for me, thank you!

**Comment by 00always** [comment]
Excuse me,How to solve the problem when executing wsl --unregister Legacy%localappdata%\lxss\rm -Recurse $env:localappdata/lxss/ in powershell. Error code: Wsl/WSL_E_WSL_OPTIONAL_COMPONENT_REQUIRED

**Reply by 00always** [reply]
Control Panel -> Programs & Features -> Turn WIndows Feature On or Off -> Windows subsystem for linux click to select, but when the computer restarts it tells me that the changes cannot be completed and the computer undoes the changes.Control Panel -> Programs & Features -> Turn WIndows Feature On or Off -> Windows subsystem for linux click to select, but when the computer restarts it tells me that the changes cannot be completed and the computer undoes the changes.

**Reply by junn-hope** [reply]
@00always, did you find a solution to this?
I am also experiencing the same issues repeatedly.

**Comment by 00always** [comment]
No, I've given up on solving this problem. I'm sorry that I can't help you.




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "microsoft/WSL"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年9月10日(星期二) 凌晨0:53
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [microsoft/WSL] how to fix  Wsl/WSL_E_WSL_OPTIONAL_COMPONENT_REQUIRED (Discussion #10074)





 
@00always, did you find a solution to this?
 I am also experiencing the same issues repeatedly.

**Comment by teddyfresnes** [comment]
Hello got the problem this earlier here is how I solved it according to last comments : 
1. doing `wsl --install --distribution kali-linux` and getting the error
2. In Windows 10 : Control Panel -> Programs & Features -> Turn WIndows Feature On or Off ->Turn on **Windows subsystem for linux** and **Hyper-V**
3. restart pc
4. reinstalled with `wsl --install --distribution kali-linux`
and it worked

**Comment by ebhrz** [comment]
seems because there is an update waiting to be installed. After I click update and restart rather than restart, it works.
